Port 3360 azure pipeline docs (#250)
port-labs/port-docs: docs/self-service-actions/azure-pipeline/examples/examples.md

---
sidebar_position: 1
---

import FindCredentials from "../../../build-your-software-catalog/sync-data-to-catalog/api/\_template_docs/\_find_credentials_collapsed.mdx";

# Examples

## Run a deployment for a service

In the following guide, you are going to build a self-service action in Port that executes an Azure pipeline behind the scenes.

### Prerequisites

- You will need your [Port credentials](../../../build-your-software-catalog/sync-data-to-catalog/api/api.md#find-your-port-credentials) to create the action;
- You will need your Azure DevOps organization;
- You will need the name of the webhook that you configured in your Azure pipelines yaml;

:::tip
<FindCredentials />
:::

### Create an Azure pipeline

First, we need to set up an Azure pipeline that implements our business logic to create a new deployment.

Here is a skeleton for the workflow - `deploy.yml` (including a placeholder for deployment logic):

<details>
<summary>Deployment Azure pipeline</summary>

```yaml showLineNumbers
trigger: none

resources:
  webhooks:
    - webhook: { WEBHOOK_NAME }
      connection: { SERVICE_CONNECTION_NAME }
stages:
  # ADD YOUR DEPLOYMENT LOGIC HERE!
```

</details>

> You can view [here](../../../build-your-software-catalog/sync-data-to-catalog/api/#generate-an-api-token) how to generate an API token to interact with Port.

Reviewer: ```suggestion
To learn how to generate an API token to interact with Port, look [here](../../../build-your-software-catalog/sync-data-to-catalog/api/#generate-an-api-token).
```

Reviewer: Move this line to what is currently line 121
